pkg/minikube/cluster/cluster.go

```diff
@@ -222,10 +223,16 @@ func createDriverOptions(driver drivers.Driver, explicitOptions map[string]inter
 	return checkFlags, nil
 }
 
+// CacheMinikubeISOFromURL download minishift ISO from a given URI.
+// It also checks sha256sum if present and then put ISO to cached directory.
 func (m *MachineConfig) CacheMinikubeISOFromURL() error {
 	fmt.Println(fmt.Sprintf("Downloading ISO '%s'", m.MinikubeISO))
+	tmpISOFile, err := ioutil.TempFile(filepath.Join(constants.Minipath, "cache", "iso"), "minishift.part")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpISOFile.Name())
```

Comment:
If you do the '.part' approach you don't want to delete the tmp file. You literally rename. See also https://github.com/minishift/minishift/blob/83e6821864e371e6ab968e950e4a04ccd489083b/pkg/minishift/docker/image/image_handler.go#L246-L246